### PR TITLE
Feat/1-song-database-model

### LINKED ISSUE #1: Song Database Model
Develop the "Song" database model using Sequelize to store song details, including title, artist, album, release year, and duration. Implement validation rules and integration into the application's database configuration.

1. Create in db/models/song.js
2. Import into db/models/index.js and export in the module.exports (this is our central hub for importing models, we will make associations here)
3. Reference image below
4.  [Create some validations for properties](https://sequelize.org/docs/v6/core-concepts/validations-and-constraints/)

Image

### PR DESCRIPTION
**Pull request by Omozax**
# Pull Request Template for Soundgarden API

## 🌟 PR Checklist

Thank you for contributing to Soundgarden API! To help us process your PR efficiently, please ensure the following:

- [x] I have read the [Contributor Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).
- [x] I have checked if there are no other [open Pull Requests](https://github.com/khalil0525/SoundGarden-API/pulls) addressing the same issue.
- [x] I have performed a self-review of my code and followed coding standards.
- [x] My changes generate no new warnings or errors.
- [x] I have added or updated necessary documentation (if appropriate).
- [x] The commit message follows our commit conventions. 💻
- [x] My branch name is descriptive, following the [branch naming guidelines](https://deepsource.com/blog/git-branch-naming-conventions).

## 📖 PR Description

This Pull Request introduces a new database model for songs, enhancing the Soundgarden API functionality. The `Song` model is designed to store essential information about songs, such as title, duration, file details, genre, privacy settings, and user ownership.

### Changes Made
- Created a new Sequelize model named `Song` in the `song.js` file.
- Defined various attributes for the `Song` model, including title, duration, file details, genre, and privacy settings.
- Implemented validation rules to ensure data integrity and consistency.
- Added an index to optimize queries based on user and permalink.
- Enhanced the uniqueness constraints for user and permalink combinations.

### Related Issues
Closes #1

## 🔧 Type of Change

- [ ] 🐛 Bug fix (A change that fixes an issue)
- [x] ✨ New feature (A non-breaking change which adds functionality)
- [ ] 💥 Breaking change (A fix or feature that would cause existing functionality to change)
- [ ] 🚀 Performance enhancement
- [ ] 🎨 Code style and formatting
- [ ] 📚 Documentation update
- [ ] 🧪 Tests (Adding missing tests or correcting existing tests)

## 🤔 Need Help?

---

Thank you for your contribution to Soundgarden API! 🎶🌱

### CODE REVIEW

File: db/models/index.js [hunk 1]
```diff
@@ -2,13 +2,20 @@
...
-const User = require('./user');
+// * 1 - 1
+
+const User = require("./user");
+const Song = require("./song");
+
+User.hasMany(Song, {
+  foreignKey: "userId",
+});
+Song.belongsTo(User);
```
Comment (khalil0525) [inline db/models/index.js:16]: 💯

File: db/models/song.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      allowNull: false,
+      validate: {
+        isUrl: true,
+      },
+    },
+    genre: {
+      type: Sequelize.STRING,
+      allowNull: false,
+    },
+    photoUrl: {
```
Comment (khalil0525) [inline db/models/song.js]: Small typo, the schema has it as "photoURL"

**Review by khalil0525 — CHANGES_REQUESTED** [review]
Great job, just one minor bug to fix and you're golden! 🎉

File: db/models/song.js (hunk 2 above)
Comment (Omozax) [reply db/models/song.js]: done

**Review by khalil0525 — APPROVED** [review]
Great job, ready to merge! 💪

**Pull request merged**